lib/semantics/resolve-names.cc

```diff
@@ -3398,6 +3400,40 @@ void DeclarationVisitor::Post(const parser::CommonBlockObject &x) {
   details->set_commonBlock(*commonBlockInfo_.curr);
 }
 
+bool DeclarationVisitor::Pre(const parser::EquivalenceStmt &x) {
+  // save equivalence sets to be processed after specification part
+  for (const std::list<parser::EquivalenceObject> &set : x.v) {
+    equivalenceSets_.push_back(&set);
+  }
+  return false;  // don't implicitly declare names yet
+}
+
+void DeclarationVisitor::CheckEquivalenceSets() {
+  EquivalenceSets equivSets{context()};
+  for (const auto *set : equivalenceSets_) {
+    const auto &source{set->front().v.value().source};
+    if (set->size() <= 1) {  // R871
+      Say(source, "Equivalence set must have more than one object"_err_en_US);
```

Comment:
Could be a warning, it's benign to have a singleton.

Reply:
OK